innerprod/fullysec/lwe_test.go

```diff
@@ -71,19 +71,21 @@ func TestFullySec_LWE(t *testing.T) {
 	assert.Error(t, err)
 	_, err = fsLWE.Decrypt(cipher, zY, emptyVec)
 	assert.Error(t, err)
-	//_, err = fsLWE.Decrypt(cipher, zX, y.MulScalar(big.NewInt(2)))
-	//assert.Error(t, err) // boundary violation
+	_, err = fsLWE.Decrypt(cipher, zY, y.MulScalar(big.NewInt(1000)))
+	assert.Error(t, err) // boundary violation
 	xyDecrypted, err := fsLWE.Decrypt(cipher, zY, y)
 	assert.NoError(t, err)
-	assert.Equal(t, xy, xyDecrypted, "obtained incorrect inner product")
+	assert.Equal(t, xy.Cmp(xyDecrypted), 0, "obtained incorrect inner product")
```

Comment:
Good catch. Do we compare two `big.Int`s with `assert.Equal` anywhere else?

Reply:
Actually we do asset.Equal in all the test.

Reply:
Hm, did this assertion give you any problems? I mean did the tests fail when they should have succeeded, or vice versa? From the docs for `assert.Equal`:
> // Pointer variable equality is determined based on the equality of the
// referenced values (as opposed to the memory addresses)

So if `xy` and `xyDecrypted` don't point to values that are equal, the test will fail. Of course, if this was not test code, you always ought to compare values with `Cmp`. But I think in this case, it only adds verbosity - especially since  `assert.Equal` and `Cmp` both compare values, and the `assert` package has our back :slightly_smiling_face: .

I'd stick to `assert.Equals(t, xy, xyDecrypted, ...)` unless I am wrong and this resulted in unexpected test results.